### DISCUSSION TITLE
[archived] "Conditional Artifacts and Parameters" docs discussion

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#13081] [General]
# en/latest/conditional-artifacts-parameters/

https://argo-workflows.readthedocs.io/en/latest/conditional-artifacts-parameters/

**Comment by djquiambao18** [comment]
What kind of syntax does the valueFrom expression follow?

**Reply by djquiambao18** [reply]
it seems to be following this: https://expr-lang.org/docs/language-definition per the docs

**Reply by djquiambao18** [reply]
However i previously encountered that some expressions like Nil coalescing is not recognized

**Reply by agilgur5** [reply]
There's certain legacy cases where Argo is doing the replacing (they predate `expr` being used) where nil coalescing doesn't work, see #11549 for more details
